package.json

```diff
@@ -15,7 +16,7 @@
   "private": false,
   "scripts": {
     "dev": "vite",
-    "build": "tsc && vite build",
+    "build": "vite build && tsc",
```

Comment:
My only question is this change. I believe I know why you swapped these 2, my concern is I don't think we should even run the build if tsc fails.

Reply:
Yeah, that's fair. I'll look into it more - for some reason if I run `tsc` before the build it won't generate the declaration files even though the `tsconfig` is set up to do so

Reply:
That's annoying! Not a blocker, just not ideal. Let me know what your research reveals.

Reply:
[Vite uses esbuild to transpile Typescript instead of `tsc`](https://vitejs.dev/guide/features.html#typescript), so I think in our build script `vite build` is overwriting the output of `tsc`. And we're not getting the emitted declaration files from `vite build` because [esbuild doesn't support it](https://github.com/evanw/esbuild/issues/95). 

So it actually seems that running `vite build` for Typescript transpilation then running `tsc` separately (with `declaration` and `emitDeclarationOnly` set to `true`) to generate the declaration files is currently the best way to go about getting what we want.

Reply:
> [Vite uses esbuild to transpile Typescript instead of `tsc`](https://vitejs.dev/guide/features.html#typescript), so I think in our build script `vite build` is overwriting the output of `tsc`. And we're not getting the emitted declaration files from `vite build` because [esbuild doesn't support it](https://github.com/evanw/esbuild/issues/95).
> 
> So it actually seems that running `vite build` for Typescript transpilation then running `tsc` separately (with `declaration` and `emitDeclarationOnly` set to `true`) to generate the declaration files is currently the best way to go about getting what we want.

Ok, nice. So if there were an error from TS it would occur within the `vite build` script. Then `tsc` would be redundant with the exception of emitting declaration files.

Reply:
Yep, that's my understanding